apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx

```diff
@@ -132,7 +132,7 @@ export const CreateOrder = () => {
       const orderResult: Swap.CreateOrderData | undefined = makePossibleMarketOrder(
         orderDetails.sell,
         orderDetails.buy,
-        orderDetails.pools,
+        [orderDetails.selectedPool],
```

Comment:
What's the point of `makePossibleMarketOrder` expecting a list of pools instead of just one, if in the only place we call it we only pass an array with one?

I see 3 options:
1. Keep this as it was
2. Modify `makePossibleMarketOrder` to expect use the one
3. Keep this change but justify in which other case we would call it with the whole list

Reply:
we had a bug here as basically when creating `makePossibleMarketOrder ` we already knew what pool was selected by the user, no point in sending the list again and filtering by the best pool again.. before we were sending the list and in some cases was not filtering right so the user was choosing a list -> but we were sending all the pool lists again and something weird was happening

Reply:
I will assume we can go with the second one if needed

Reply:
this is the api to make a possible market order, which means that it should only operate on the pools that we support, I'd say if "none" is provided, we should look into all the supported pools, if we provide we provide one only.

Reply:
as long as we provide pools in the support list, I think it's fine.

Reply:
I don't think this function can be called without a selectedPool present. And the selectedPool should already be a supported one, because they get filtered as soon as we get them. So I would say refactor `makePossibleMarketOrder` to remove the selection from array.

Reply:
k, I think it's a little bit more complex, I think we should ask product, cuz if the user didn't touch the selected pool, and is trying to create a market order, I'd say it should check all supported pools, get the best price, and then file the order. But if the user selected a pair, and has selected a specific pool, I'm assuming that pool is the one wanted, so we should create an order for that pool.

Reply:
yes, if the user is not touching the selected pool -> the default one with the best price will be used to create the order and if is choosing a specific one, that will be used.  No point in sending the list again when `makePossibleMarketOrder` as we already know the selected by default has the best price. 
- If we are sending the list when creating the  `makePossibleMarketOrder` again, even if user is choosing a specific pool - then the function will sort them and choose the best one anyway 🙃

Reply:
for not touching the helper and calculation have just sent the best one in the array as an easy fix - but I think the best approach will be to refactor the `makePossibleMarketOrder` to always receive the default one or the one user chooses to create a order